EX2/turtle_sent7.2.py

```diff
@@ -0,0 +1,121 @@
+                    urgent (bool, optional): The urgency of the message.
+                                            Urgent messages appear first.
+
+                Returns:
+                    int: The message ID, auto incremented number.
+
+                Raises:
+                    KeyError: If the recipient does not exist.
+
+                Examples:
+                    After creating a PO box and sending a letter,
+                    the recipient should have 1 message in the
+                    inbox.
+
+                """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'title': message_title,
+            'body': message_body,
+            'sender': sender,
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+    def read_inbox(self, username, n=0):
+        """Read n messages from the user inbox.
+
+                Args:
+                    username (str): The username.
+                    n (int): The number of messages to read, default = 0 - read all the inbox.
+
+                Returns:
+                    list: The first n messages from the user inbox to read.
+
+                Raises:
+                    KeyError: If the username does not exist.
+
+                Examples:
+                    After reading 1 message,
+                    if inbox had 2 messages and history was empty,
+                    now the username should have only the second message in the inbox
+                    and the first inbox message in history.
+
+                """
+        user_box = self.boxes[username]
+        if n == 0:
+            n = len(user_box)
+
+        msgs_to_read = user_box[:n]
+        self.history[username].append(msgs_to_read)
+        self.boxes[username] = user_box[n:]
+        return msgs_to_read
+
+    def search_inbox(self, username, text):
+        """Read all messages that contains the text.
+
+                Args:
+                    username (str): The username.
+                    text (str): The text the message needs to contain.
+
+                Returns:
+                    list: All the messages from the user inbox that contains the text
+                    in their title or in thier body.
+
+                Raises:
+                    KeyError: If the username does not exist.
+
+                Examples:
+                    After reading 1 message,
+                    if inbox had 2 messages and history was empty,
+                    now the username should have only the second message in the inbox
+                    and the first inbox message in history.
+
+                """
+        return [msg for msg in self.boxes[username] if text in msg['body'] or text in msg['title']]
+
+
+def main():
+    po_box = PostOffice(['a', 'b'])
```

Comment:
Don't use abbreviations in variable names -
https://softwareengineering.stackexchange.com/questions/72593/how-to-abbreviate-variable-names